src/utils/Validator.ts

```diff
@@ -1,26 +1,30 @@
 export class Validator {
-    // tslint:disable-next-line: no-empty
-    public constructor() {}
+  // tslint:disable-next-line: no-empty
+  public constructor() {}
 
-    /**
-     * Validate query or path parameter provided in the request
-     * @returns boolean Result of logic determining validity
-     * @param parameters This is expecting an object as a parameter
-     */
-    public parametersAreValid<T extends string>(parameters: T): boolean {
-        let isValid: boolean = true;
-        for (const key of Object.keys(parameters)) {
-            const value: string = Object.values(parameters)[key.indexOf(key)];
-            if (value) {
-                if (value.trim().length === 0 || value === "undefined" || value === "null") {
-                    isValid = false;
-                    break;
-                }
-            } else {
-                isValid = false;
-                break;
-            }
+  /**
+   * Validate query or path parameter provided in the request
+   * @returns boolean Result of logic determining validity
+   * @param parameters This is expecting an object as a parameter
+   */
+  public parametersAreValid<T extends string>(parameters: T): boolean {
```

Comment:
Tech records team is using [joi](https://www.npmjs.com/package/joi) for validation.
How easy/difficult it is to apply the same here? 
what do you think @Ian-Hodges ?

Reply:
There are no logical changes in the validation as part of this story/PR, the changes were purely formatting as a result of the prettier run. 

Joi validation is already used in this service, but similarly with tech records it is used to validate the values of the query parameters posted and not the path parameters.

I think implementing for path parameters would be more work, as presumably it would require re-working/removing the logic above and subsequnetly amending/creating schemas - but if you feel this is a piece of work that needs to be done, then I could speak to our BA and get some stories created in the backlog, as it would likely mean changes to the majority of backend services.

Reply:
Thank you for your response.
I would recommend adapting the same validation technique across the board but this should be implemented as you work on a ticket rather than create one ticket to change all validations in all back end services.
Having a quick look at at how joi works, the validate parameter function could receive an object call queryParams for example so
`
const lambdaHandler = (request) => {
const isQueryParamsValid = validateQueryParam({paramOne: request.queryParams.paramOne});
if(!isQueryParamsValid){
throw ...etc
}
}
`
Now in validateQueryParam function you can easily create a schema to validate this object using joi.
Hope this make sense. Please let me know your thoughts.

Reply:
I think we can add enhancements as we go along, as in applying the boy-scout rule (Always leave the code better than you found it).
If you don't want to implement the change in this story (because of scope for example) then that's your call. My suggestions are not compulsory by any shape or form. I don't think you need to block the story in JIRA

Reply:
Thanks for the comments, I do appreciate where you are coming from - particularly with regards to leaving the code better than the way it was found. I think if this story itself was manipulating the validation then I agree it would be a great time to update as per your suggestion, that would also mean the changes would be thoroughly tested by QA as part of that story. I would just be reluctant to change whilst it is out of scope.

I'd be keen to get the thoughts of my manager, so will certainly make a note of this and discuss further with her when she returns to work.